relay-server/src/services/store.rs

```diff
@@ -1378,16 +1378,15 @@ struct SpanKafkaMessage<'a> {
     /// Number of days until these data should be deleted.
     #[serde(default)]
     retention_days: u16,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    segment_id: Option<&'a str>,
+    segment_id: &'a str,
```

Comment:
We deserialize a `SpanKafkaMessage` from a `Span`, right? Can we be sure that it has a `segment_id`?

Reply:
Reverted since we're using `trace_id` and not `segment_id` for routing.